src/apis/free-post-comments/services/free-post-comments.service.ts

```diff
@@ -84,8 +84,18 @@ export class FreePostCommentsService {
 
     const relations = this.queryHelper.createNestedChildRelations(loadDepth);
 
+    /**
+     * @todo 1 이상 depth도 처리되게 변경
+     * @todo join 후 where 필터링이 아닌 join on 조건으로 필터링되게
+     */
     return this.freePostCommentRepository.findAndCount({
-      where: { ...where, depth: 0 },
+      where: {
+        ...where,
+        depth: 0,
+        children: {
```

Comment:
개인적인 생각인데 TypeORM의 DeleteDateColumn 데코레이터를 사용해보는 것은 어떨까요?
@DeleteDateColumn
Special column that is automatically set to the entity's delete time each time you call soft-delete of entity manager or repository. You don't need to set this column - it will be automatically set.
TypeORM's own soft delete functionality utilizes global scopes to only pull "non-deleted" entities from the database.
If the @DeleteDateColumn is set, the default scope will be "non-deleted".
```typescript
@Entity()
export class User {
    @DeleteDateColumn()
    deletedDate: Date
}
```

Reply:
처음에 soft delete로 바꾸면서 저걸 쓸까도 고려했는데 status 를 함께 관리하는게 안돼서 안썻습니다.

Reply:
softDelete나 softRemove 메서드를 쓰지 않고 update 메서드를 통해 deletedAt을 갱신시켜 주어도 find 메서드등을 사용할때 자동으로 deletedAt IS NULL 조건을 걸어줍니다

Reply:
softDelete나 softRemove 메서드를 쓰지 않고 update 메서드를 통해 deletedAt을 갱신시켜 주어도 find 메서드등을 사용할때 자동으로 deletedAt IS NULL 조건을 걸어줍니다

Reply:
softDelete나 softRemove 메서드를 쓰지 않고 update 메서드를 통해 deletedAt을 갱신시켜 주어도 find 메서드등을 사용할때 자동으로 deletedAt IS NULL 조건을 걸어줍니다. 상관없지 않을까요?

Reply:
ㅁㄴㅇ

Reply:
코멘트 테스트

Reply:
오 그런가요??
그러면 해당 부분은 따로 검토해서 전체적으로 적용할지 논의해보먄 좋을듯합니다
우선 이번 pr 에는 적용하지 않으려하는데 괜찮을까요??

Reply:
넵 그 방향이 더 좋을 것 같네요!